lib/repos/git/fetch.go

```diff
@@ -19,8 +21,29 @@ import (
 
 var debug = false
 
+func IsGit(ctx context.Context, host, owner, repo string) (bool, error) {
```

Comment:
I want to think about this a bit more. Should we do the direct calls to gogit here? In most cases we want to avoid a network call. We should look in the cache first if possible. Maybe the cache should do the determination and we have one func that takes the params and returns the type? (rather than distributing the decision?)